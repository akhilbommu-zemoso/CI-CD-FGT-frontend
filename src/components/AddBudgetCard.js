import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { FTCard } from '../atoms/FTCard';
const useStyles = makeStyles({
    cardStyle: {
        width: '200px',
        fontWeight: 900,
        borderRadius: '6px',
        border: '1px,solid,black',
        color: 'rgb(22,101,216)',
        backgroundColor: 'rgb(255, 255, 255)',
        textAlign: 'center',
        '&:hover': {
            backgroundColor: "rgb(4,136,255)",
            color: 'white'
        }
    },
    content: {
        marginTop: "14px",
        fontSize: 12,
        textAlign: 'center',
        fontWeight: 500,
        color: 'black',
        textAlign: 'center',
    },
});
export default function Budgetcard(props) {
    const classes = useStyles();
    return (
        <FTCard
            cardStyle={classes.cardStyle}
            contentStyle={classes.content}
            headerStyle={classes.header}
            header={"Create your budget"}
            content={"Take control on your expenses by creating a budget on your spendings"}
        />
    );
}