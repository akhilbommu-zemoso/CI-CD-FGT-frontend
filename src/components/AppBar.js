import React from 'react';
import { fade, makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import Badge from '@material-ui/core/Badge';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import AccountCircle from '@material-ui/icons/AccountCircle';
import NotificationsIcon from '@material-ui/icons/Notifications';
import MoreIcon from '@material-ui/icons/MoreVert';
import HelpOutlineSharpIcon from '@material-ui/icons/HelpOutlineSharp';
import ProfileList from './ProfileList'
import Brand from '../atoms/Brand'
import ProfilePicture from './ProfilePicture';
const useStyles = makeStyles((theme) => ({
    grow: {
        maxWidth: `calc(100%-240px)`,
    },
    menuButton: {
        marginRight: theme.spacing(2),
    },
    title: {
        marginTop: '10px',
        fontWeight: 700,
        [theme.breakpoints.up('sm')]: {
            display: 'block',
        },
    },
    search: {
        position: 'relative',
        borderRadius: theme.shape.borderRadius,
        border: '1px solid #dddddd',
        backgroundColor: fade(theme.palette.common.white, 0.15),
        '&:hover': {
            backgroundColor: fade(theme.palette.common.white, 0.25),
        },
        marginRight: theme.spacing(2),
        marginLeft: 0,
        width: '100%',
        [theme.breakpoints.up('sm')]: {
            marginLeft: theme.spacing(3),
            width: 'auto',
        },
    },
    searchIcon: {
        padding: theme.spacing(0, 2),
        height: '100%',
        position: 'absolute',
        pointerEvents: 'none',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
    },
    inputRoot: {
        color: 'inherit',
    },
    inputInput: {
        padding: theme.spacing(1, 1, 1, 0),

        paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
        transition: theme.transitions.create('width'),
        width: '100%',
        [theme.breakpoints.up('md')]: {
            width: '20ch',
        },
    },
    sectionDesktop: {
        display: 'none',
        [theme.breakpoints.up('md')]: {
            display: 'flex',
        },
    },
    sectionMobile: {
        display: 'flex',
        [theme.breakpoints.up('md')]: {
            display: 'none',
        },
    },
    red: {
        color: 'black',
        backgroundColor: 'rgb(255,255,255)'
    },
    space: {
        display: 'flex',
        justifyContent: 'space-between',
        width: '100%',
    },
    linksclass: {
        color: 'black',
        paddingTop: '0px',
        textDecoration: 'none',
        '&:hover': {
            color: 'black',
            textDecoration: 'none',
        },
    },
    logo: {
        width: '30px',
        height: '30px',
        fill: 'white',
        backgroundColor: "rgb(4,136,255)",
        borderRadius: "50%",
        padding: '5px',
        marginBottom: '6px',
        marginRight: '10px',
    },
    padding: {
        paddingLeft: '10px',
        paddingRight: '20px',
    },
    nopadding: {
        padding: '0px',
    },
    avatar: {
        backgroundColor: 'rgba(182,32,224,0.71)',
        width: '30px',
        height: '30px',
        borderRadius: '50%',
        fontSize: '18px'
    },
    iconpadding: {
        paddingLeft: '15px',
        paddingRight: '15px',
    }


}));
export default function PrimarySearchAppBar(props, children) {
    const classes = useStyles();
    const [anchorEl, setAnchorEl] = React.useState(null);
    const [mobileMoreAnchorEl, setMobileMoreAnchorEl] = React.useState(null);
    const isMenuOpen = Boolean(anchorEl);
    const isMobileMenuOpen = Boolean(mobileMoreAnchorEl);
    const handleProfileMenuOpen = (event) => {
        setAnchorEl(event.currentTarget);
    };
    const handleMobileMenuClose = () => {
        setMobileMoreAnchorEl(null);
    };
    const handleMenuClose = () => {
        setAnchorEl(null);
        handleMobileMenuClose();
    };
    const handleMobileMenuOpen = (event) => {
        setMobileMoreAnchorEl(event.currentTarget);
    };
    const menuId = 'primary-search-account-menu';
    const renderMenu = (
        <Menu
            data-testid='menuclose'
            MenuListProps={{ disablePadding: true }}
            anchorEl={anchorEl}
            anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
            id={menuId}
            keepMounted
            transformOrigin={{ vertical: 'top', horizontal: 'right' }}
            open={isMenuOpen}
            onClose={handleMenuClose}
        >
            {/* rendering profile list when user clicks on appbar avtar */}
            {props.profilelist}
        </Menu>
    );
    const mobileMenuId = 'primary-search-account-menu-mobile';
    const renderMobileMenu = (
        <Menu
            data-testid='mobileclose'
            anchorEl={mobileMoreAnchorEl}
            anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
            id={mobileMenuId}
            keepMounted
            transformOrigin={{ vertical: 'top', horizontal: 'right' }}
            open={isMobileMenuOpen}
            onClose={handleMobileMenuClose}
        >
            <MenuItem>
                <IconButton aria-label="show  new mails" color="inherit">
                    <Badge color="secondary">
                        <HelpOutlineSharpIcon />
                    </Badge>
                </IconButton>
                <p>Messages</p>
            </MenuItem>
            <MenuItem>
                <IconButton aria-label="show  new notifications" color="inherit">
                    <Badge color="secondary">
                        <NotificationsIcon />
                    </Badge>
                </IconButton>
                <p>Notifications</p>
            </MenuItem>
            <MenuItem onClick={handleProfileMenuOpen}>
                <IconButton
                    aria-label="account of current user"
                    aria-controls="primary-search-account-menu"
                    aria-haspopup="true"
                    color="inherit"
                >
                    <AccountCircle />
                </IconButton>
                <p>Profile</p>
            </MenuItem>
        </Menu>
    );
    return (
        <div className={classes.grow}>
            <AppBar position="fixed" className={classes.red}>
                <Toolbar className={classes.padding}>
                    <div className={classes.space} >
                        <div data-testid="brand">
                            {/* rendering the brand ZE-Transact */}
                            {props.brand}
                        </div>
                        <div >
                            <div className={classes.sectionDesktop}>
                                <IconButton aria-label="show 4 new mails" color="inherit" className={classes.iconpadding}>
                                    <Badge color="secondary" data-testid="help">
                                        <HelpOutlineSharpIcon />
                                    </Badge>
                                </IconButton>
                                <IconButton aria-label="show  new notifications" color="inherit" className={classes.iconpadding}>
                                    <Badge color="secondary" data-testid="notification">
                                        <NotificationsIcon />
                                    </Badge>
                                </IconButton>
                                <IconButton
                                    data-testid="iconbutton"
                                    edge="end"
                                    aria-label="account of current user"
                                    aria-controls={menuId}
                                    aria-haspopup="true"
                                    onClick={handleProfileMenuOpen}
                                    color="inherit"
                                >
                                    <div data-testid="picture">
                                        <ProfilePicture pictureStyle={classes.avatar} />
                                    </div>
                                </IconButton>
                            </div>
                        </div>
                    </div>
                    <div className={classes.sectionMobile}>
                        <IconButton
                            data-testid="mobilemenu"
                            aria-label="show more"
                            aria-controls={mobileMenuId}
                            aria-haspopup="true"
                            onClick={handleMobileMenuOpen}
                            color="inherit"
                        >
                            <MoreIcon />
                        </IconButton>
                    </div>
                </Toolbar>
            </AppBar>
            {renderMobileMenu}
            {renderMenu}
        </div>
    );
}
