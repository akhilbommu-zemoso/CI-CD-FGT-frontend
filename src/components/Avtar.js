import React, { useState, useEffect } from "react";
import { makeStyles, withStyles, fade } from "@material-ui/core/styles";
import { FormControl, InputLabel, Input, InputBase } from "@material-ui/core";
import Avatar from "@material-ui/core/Avatar";

import { isLogin } from '../utils/index'
const useStyles = makeStyles({
  circle: {
    height: "150px",
    width: "150px",
    backgroundColor: "rgba(182,32,224,0.71)",
    borderRadius: "50%",
    fontSize: 50,
  },
});
export default function Avtar(props) {
  const classes = useStyles();
  const uploadedImage = React.useRef(null);
  const imageUploader = React.useRef(null);
  const [initialSetUp, setInitialSetUp] = useState();
  const [picObject, setPicObject] = useState({
    id: 0,
    photoBlob: null,
    photoContentLength: 0,
    photoContentType: null,
  });
  const handleImageUpload = async (event) => {
    const [file] = event.target.files;
    const { current } = uploadedImage;
    const reader = new FileReader();
    if (file && file.type.match("image.*")) {
      reader.readAsDataURL(file);
      reader.onload = async (event) => {
        getData(event.target.result, file.type).then((data) => {
          setPicObject(data);
          props.handleChangePicture(data);
        });
      };
    }
  };
  const getData = async (result, fileType) => {
    let photoData = result.split(",")[1];
    let temp = { ...picObject };
    temp.photoContentType = fileType;
    temp.photoContentLength = photoData.length;
    temp.photoBlob = await photoData;
    return temp;
  };
  useEffect(() => {
    picObject.id=props.imageData.id;
    if (props.imageData.photoContentLength) {
      setPicObject(props.imageData);
    }
  }, [initialSetUp]);
   
  useEffect(() => {
    console.log(picObject.photoContentLength);
    if (picObject.photoContentLength) {
      const { current } = uploadedImage;
      current.src = "data:"+picObject.photoContentType+";base64," + picObject.photoBlob;
    }
  }, [picObject]);
  return (
    <div>
      <input
        id="inputfile"
        data-testid="avtarinput"
        type="file"
        accept="image/*"
        onChange={handleImageUpload}
        ref={imageUploader}
        style={{ display: "none" }}
      />
      <div data-testid="uploadclick" onClick={() => imageUploader.current.click()}>
        {picObject.photoContentLength ? (
          <div>
            <img ref={uploadedImage} className={classes.circle} />
          </div>
        ) : (
          <Avatar aria-label="label" className={classes.circle}>
            {isLogin() && JSON.parse(localStorage.getItem('user')).name.substring(0,2).toUpperCase()}
          </Avatar>
        )}
      </div>
    </div>
  );
}