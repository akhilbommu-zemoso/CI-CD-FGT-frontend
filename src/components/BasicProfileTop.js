import React, { useState } from "react";
import { makeStyles, withStyles, fade } from "@material-ui/core/styles";
import { FormControl, InputLabel, Input, InputBase, Select, MenuItem,TextField } from "@material-ui/core";
import CountrySelect from "./CountryField"
import BootstrapInput from '../atoms/TextField'
import ErrorField from '../atoms/ErrorField'
const useStyles = makeStyles((theme) => ({
    alignField: {
        paddingBottom: "20px"
    },
    labelSize: {
        fontSize: "20px",
        color: "rgb(72,84,96)",
    },
    formControl: {
        backgroundColor: "white",
        fontSize: 16,
        width: "350px",
    },
    circle: {
        height: "150px",
        width: "150px",
        backgroundColor: "pink",
        borderRadius: "50%"
    }
}));
function ProfileFormFields({ values, handleChange,handleChangeCountry,errorMessage }) {
    const classes = useStyles();
    return (
        <div>
            <div className={classes.alignField}>
                <FormControl className={classes.margin}>
                    <InputLabel shrink htmlFor="name" className={classes.labelSize} >
                        User Name
                   </InputLabel>
                <BootstrapInput data-testid="input" name="name" value={values.name} id="name" onChange={(event) => handleChange(event)} autoComplete="off" />
                </FormControl>
                <ErrorField message={errorMessage.name}/>
            </div>
            <div className={classes.alignField}>
                <FormControl className={classes.margin}>
                    <InputLabel shrink htmlFor="phone" className={classes.labelSize}>
                        Phone Number
        </InputLabel>
                    <BootstrapInput name="phone" inputProps={{ pattern: "[a-z]" }} required value={values.phone} id="phone" onChange={(event) => handleChange(event)} autoComplete="off"/>
                </FormControl>
                <div>
                    <ErrorField message={errorMessage.phone}/>
                    </div>
            </div>
            <CountrySelect country={values.country}
                handleChangeCountry={(event)=>handleChangeCountry(event)}
            />
        </div>
    );
}
export { ProfileFormFields}