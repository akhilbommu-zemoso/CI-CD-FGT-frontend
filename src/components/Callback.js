import React, { useEffect, useState } from 'react'
import { Redirect} from 'react-router-dom'
import { login } from '../utils/index'
function Callback(props) {
    const [loading, setLoading] = useState(false)
    const [customError, setcustomError] = useState(null);
    const loginApi = async (accessToken) => {
        var apiurl = process.env.REACT_APP_BACKEND_USER;
        console.log(accessToken);
       fetch(apiurl, {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': 'Bearer ' + accessToken,
            },
        }).then(response => {
            if(!response.ok){
                throw new Error(response.status)
            }
            return response.json();
        })
            .then((data)=>{return data;})
           .then((data) => login(data, accessToken))
            .then(()=>setLoading(true))
                .catch((error) => {
                    if(error.message==="401"){
                        setcustomError("Unauthorized Access");}
                    else if(error.message==="404"){
                        setcustomError("Invalid User");}
                    else if(error.message==="500"){
                        setcustomError("Internal Server Error");}
                    else{
                        setcustomError("Unable to fetch resource")
                    }
                    return <Redirect to="/error" />
                });
    }
    useEffect(() => {
        if (window.location.href.includes("access_token")) {
            var url = new URL(window.location.href);
            var res = url.hash.split("=");
            var accessToken = res[1].slice(0, res[1].length - 6);
            console.log(accessToken);
                loginApi(accessToken);
        }
    })
    const renderRedirect = () => {
        console.log(customError)
        if (customError) {
            const newRoute = `/error?${customError}`
            return <Redirect to={newRoute} />

        }
        else{
            return (loading&&<Redirect to="/dashboard" />)
        }
      }
    return (
        <div>
            {renderRedirect()}
        </div>
    )
}
export default Callback