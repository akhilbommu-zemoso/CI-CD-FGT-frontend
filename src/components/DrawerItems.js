import React from 'react'
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import Divider from '@material-ui/core/Divider'
import { Link } from 'react-router-dom'
function DrawerItems(props) {
    return (
        <Link data-testid="link" to={props.link} className={props.linksclass} onClick={() => { props.setSelected(2 ** props.index) }} >
            <ListItem
                selected={(props.selected >> props.index) & 1}
                classes={(props.selected:props.active)}>
                <Divider classes={{ root: props.dividerColor }}
                flexItem={(props.selected >> props.index) & 1}
                orientation="vertical" />
                 <ListItemIcon>{props.icon}</ListItemIcon>
                <ListItemText primary={props.itemName} />
            </ListItem>
        <Divider />
        </Link >
      
    )
}

export default DrawerItems