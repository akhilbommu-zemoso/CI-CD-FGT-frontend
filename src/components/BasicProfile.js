import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { ProfileFormFields } from './BasicProfileTop'
import Avtar from './Avtar'
import Divider from '@material-ui/core/Divider';
import Snackbar from '@material-ui/core/Snackbar';
import MuiAlert from '@material-ui/lab/Alert';
import FTButton from '../atoms/FTButton'
import ProfileDeleteAccount from './ProfileDeleteAccount'
import ProfileEmail from './ProfileEmail'
import ProfilePassword from './ProfilePassword'
const useStyles = makeStyles((theme) => ({
    padding: {
        padding: "20px"
    },
    colorbutton: {
        marginTop: '50px',
        color: 'white',
        textTransform: 'none',
        marginBottom: '50px',
        backgroundColor: 'rgb(4,136,255)',
        '&:hover': {
            color: 'white',
            backgroundColor: 'rgb(4,136,255)'
        },
    },
    flex: {
        display: "flex",
        justifyContent: "space-between"
    },
    alignRight: {
        width: '100%',
        textAlign: 'right'
    },
    root: {
        width: '100%',
        '& > * + *': {
            marginTop: theme.spacing(2),
        },
    }
}));

export default function BasicProfile(props) {
    const classes = useStyles();
    function Alert(props) {
        return <MuiAlert elevation={6} variant="filled" {...props} />;
      }
      const checkErrorMessage=()=>{
        if(props.errorMessage.name===""&&props.errorMessage.phone==="")
            return false;
        else
            return true;
      }
    return (
        <div>
            <div className={classes.padding}>
                <div className={classes.flex}>
                    <ProfileFormFields
                        id="profileform"
                        values={props.values}
                        handleChange={(event) => props.handleChange(event)}
                        handleChangeCountry={(event)=>props.handleChangeCountry(event)}
                        errorMessage={props.errorMessage}/>
                    <Avtar imageData={props.values.profilePicture}
                        userName = {props.values.name}
                        handleChangePicture={(data)=>props.handleChangePicture(data)}/>
                </div>
            </div>
            <Divider />
            <div className={classes.padding} data-testid="profileemail">
                <ProfileEmail email={props.values.email} />
            </div>
            <Divider />
            <div className={classes.padding} data-testid="profilepassword">
                <ProfilePassword />
            </div>
            <Divider />
            <div className={classes.padding} data-testid="profiledeleteaccount">
                <ProfileDeleteAccount />
            </div>
            <div className={classes.alignRight}>
                    <FTButton
                        data-testid="button"
                        name="Save"
                        method={(event) => props.postToUserApi()}
                        disabled={checkErrorMessage()}
                    />
                <Snackbar
                    data-testid='snackbar'
                    anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
                    open={props.open} autoHideDuration={6000}
                    onClose={(event, reason) => props.handleClose(event, reason)}>
                    <Alert onClose={(event, reason) => props.handleClose(event, reason)} severity="success">
                        Changes Saved!
                    </Alert>
                </Snackbar>
            </div>
        </div>
    );
}