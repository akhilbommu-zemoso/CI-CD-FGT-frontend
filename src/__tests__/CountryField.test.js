import React, { Component } from 'react'
import { render, fireEvent, getAllByText,screen } from '@testing-library/react'
import { renderHook, act } from '@testing-library/react-hooks'
import { BrowserRouter as Router } from 'react-router-dom'
import { ProfileFormFields, BootstrapInput } from '../components/BasicProfileTop'
import BasicProfile from '../components/BasicProfile'
import CountrySelect from '../components/CountryField'
const handleChange = jest.fn();
const handleChangeCountry = jest.fn();
const data = () => {
    var json = {
        id: "1",
        name: "Fareed Sheik",
        email: "sheikfareed3819@gmail.com",
        auth0Id: "google - oauth2 | 115320575601904541142",
        phone: " + 91 - 7680024324", "country": "United Arab Emirates",
        createdAt: "1589626102282", "createdBy": "Fareed",
        modifiedAt: "1589626102282", "modifiedBy": "Fareed",
        country:"India",
        profilePicture: {
            id: "1",
            photoBlob: null,
            photoContentLength: "0",
            photoContentType: null,
        }
    }
    localStorage.setItem('user', JSON.stringify(json))
    return json;
}

it('should take a snapshot', () => {
    const { asFragment } = render(<Router>
        <CountrySelect
            handleChangeCountry={handleChangeCountry}
            country={""}
        /></Router>
    )
    expect(asFragment(<Router>
        <CountrySelect
            handleChangeCountry={handleChangeCountry}
            country={data().country}
        />
    </Router>)).toMatchSnapshot()
});

it("checks for label", () => { 
    const { getByText,getByTestId,getByRole } = render(<Router>
        <CountrySelect
            handleChangeCountry={handleChangeCountry}
            country={"India"}
        /></Router>
    )
    expect(screen.getByText("Country")).toBeInTheDocument 
})
   
      
 
