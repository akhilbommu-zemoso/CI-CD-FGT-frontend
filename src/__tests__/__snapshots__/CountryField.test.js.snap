// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelSize-4 MuiInputLabel-animated"
      for="country"
    >
      Country
    </label>
    <div
      class="MuiFormControl-root"
    >
      <div
        aria-expanded="false"
        class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
        name="country"
        role="combobox"
      >
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-formtextField-3 MuiFormControl-fullWidth"
          style="color: red;"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <input
              aria-autocomplete="list"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="new-password"
              class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              id="country"
              spellcheck="false"
              style="font-size: 14px; padding: 3.5px 0px;"
              type="text"
              value=" "
            />
            <div
              class="MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-108 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-109"
                style="width: 0.01px;"
              >
                <span>
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
